feat(bitbucketdatacenter): Add support for fetching and converting projects to teams

Levy-Tal opened the pull request:
- Enabled WOODPECKER_ORGS in .env file to restrict CI tool access to specific organizations, enhancing administrative control.
- Implemented GetAllProjects function to retrieve paginated projects from Bitbucket Data Center.
- Added Teams function to convert Bitbucket projects into teams for integration with the Woodpecker model.
- Introduced convertProjectsToTeams helper function to transform projects into team structures with appropriate avatars.
- Added unit tests for convertProjectsToTeams in convert_test.go.


This update improves support for Bitbucket Data Center by enabling fine-grained organization access control and project-to-team mapping, essential for Woodpecker CI administrators.

<!--

Please check the following tips:
1. Avoid using force-push and commands that require it (such as `commit --amend` and `rebase origin/main`). This makes it more difficult for the maintainers to review your work. Add new commits on top of the current branch, and merge the new state of `main` into your branch with plain `merge`.
2. Provide a meaningful title for this pull request. It will be used as the commit message when this pull request is merged. Add as many commits as you like with any messages you like, they will be squashed into one commit.
3. If this pull request fixes an issue, refer to the issue with messages like `Closes #1234`, or `Fixes #1234` in the pull description.
4. Please check that you are targeting the `main` branch. Pull requests on release branches are only allowed for backports.
5. Make sure you have read contribution guidelines: https://woodpecker-ci.org/docs/development/getting-started
6. It is recommended to enable "Allow edits by maintainers", so maintainers can help you more easily.

-->

Levy-Tal commented on server/forge/bitbucketdatacenter/bitbucketdatacenter.go:
```diff
@@ -570,10 +570,47 @@ func (c *client) updatePipelineFromCommit(ctx context.Context, u *model.User, r
-func (*client) Teams(_ context.Context, _ *model.User) ([]*model.Team, error) {
-	var teams []*model.Team
-	return teams, nil
+func GetAllProjects(ctx context.Context, client *bb.Client, opts *bb.ListOptions) ([]*bb.Project, error) {
```
I merged the Teams and GetAllProjects functions

langecode commented:
> Thanks! Pinging @langecode as you're the bb datacenter responsible.

Thanks. Looks good to me. We haven't played much with the teams in Woodpecker so I did not consider mapping projects to teams - but it would probably be the best alternative as Bitbucket Datacenter doesn't really have notion of teams.

Btw, I've added a `ListProject` to the BitBucket client:
https://github.com/neticdk/go-bitbucket/blob/main/bitbucket/projects.go#L23

Levy-Tal commented:
Update the Teams function to use new ListProjects method

Levy-Tal commented:
- Removed unnecessary empty line to satisfy linter requirements.

- Replaced nil slice(`var allProjects []*bb.Project`) with empty slice (`allProjects := make([]*bb.Project, 0)`) for consistency with the rest of the codebase.
